index.js

```diff
@@ -194,4 +195,103 @@ export default async function pMap(
 	});
 }
 
+export function pMapIterable(
+	iterable,
+	mapper,
+	{
+		concurrency = Number.POSITIVE_INFINITY,
+	} = {},
+) {
+	if (iterable[Symbol.iterator] === undefined && iterable[Symbol.asyncIterator] === undefined) {
+		throw new TypeError(`Expected \`input\` to be either an \`Iterable\` or \`AsyncIterable\`, got (${typeof iterable})`);
+	}
+
+	if (typeof mapper !== 'function') {
+		throw new TypeError('Mapper function is required');
+	}
+
+	if (!((Number.isSafeInteger(concurrency) || concurrency === Number.POSITIVE_INFINITY) && concurrency >= 1)) {
+		throw new TypeError(`Expected \`concurrency\` to be an integer from 1 and up or \`Infinity\`, got \`${concurrency}\` (${typeof concurrency})`);
+	}
+
+	return {
+		[Symbol.asyncIterator]() {
+			const {next, nextError, complete, onCleanup, iterator} = createDeferredAsyncIterator();
```

Comment:
I would prefer to do this without introducing another dependency, if possible. I was hoping to make `p-map` dependency-less soon when I can remove `'aggregate-error'`. This would be beneficial as `p-map` is used a LOT.

Reply:
Yeah, I'm rewriting to add backpressure from the parent issue.
Gonna remove that then.